test/StateSchedulingTest.cpp

```diff
@@ -0,0 +1,125 @@
+#include "pipert/Scheduler.h"
+#include "pipert/ScheduledChannel.h"
+#include "pipert/PolledChannel.h"
+#include "pipert/Packet.h"
+#include "pipert/Timer.h"
+#include "gtest/gtest.h"
+#include <iostream>
+#include <unistd.h>
+
+namespace {
+
+class Base {
+ public:
+  int GetValue() const { return value_; }
+ protected:
+  int value_ = 0;
+};
+
+class Processor : public Base {
+ public:
+  void Process(::pipert::PacketToProcess<int> packet) {
+    for (int i = 0; i < 10; ++i) {
+      value_ += packet.data();
+    }
+    for (int i = 0; i < 10; ++i) {
+      value_ -= packet.data();
+    }
+  }
+};
+
+class Incrementor : public Base {
+ public:
+  Incrementor(::pipert::ScheduledChannel<int>&& decChannel) : decChannel_(std::move(decChannel)) {}
+  void Increment(::pipert::PacketToProcess<int> packet) {
+    value_ += packet.data();
+    ::pipert::PacketToFill<int> new_packet = decChannel_.Acquire("Decrementor", packet.timestamp(), packet.data());
+    while(packet.IsEmpty()) {
+      new_packet = decChannel_.Acquire("Decrementor", packet.timestamp(), packet.data());
+    }
+    new_packet.Push();
+  }
+
+  int GetDroppedPacketsNumber() {
+    return decChannel_.GetDroppedPacketsNumber();
+  }
+ private:
+  ::pipert::ScheduledChannel<int> decChannel_;
+};
+
+class Decrementor : public Base {
+ public:
+  void Decrement(::pipert::PacketToProcess<int> packet) {
+    value_ -= packet.data();
+  }
+};
+
+}  // namespace
+
+TEST(StateScheduling, SchedulingWithoutState) {
```

Comment:
How is this test functionally different from SchedulingWithState? The `client_name` parameter here "Incrementer" seems like it might be a misspell, and even then, client names are explicitly ignored at the moment if I'm right. Looks like both tests use only the Processor, and therefore they seem equivalent, which seems unintended.

What is the purpose of testing `processor.GetValue()` to be between -1000 and 1000? The value printed for me was always zero.

Could you please clarify how these two tests are supposed to work?

Reply:
In this test case the channel creation happens with a nullptr, so it's not bound to a single thread object, it means that multiple ```Processor::Process``` can run in the same time, and because the ```value_``` in the process class is not synchronized, we can't really determine the value of ```value_```. But you are right, this test case is not so verbose, and there could be better solution for checking the stateless functionality.

Reply:
Yeah, you're right, can't believe I missed that. Thank you for your explanation, I should look more into how stateful tasks are implemented.